workitem/enum_type.go

```diff
@@ -40,6 +40,14 @@ var _ convert.Equaler = (*EnumType)(nil)
 
 // Equal returns true if two EnumType objects are equal; otherwise false is returned.
 func (t EnumType) Equal(u convert.Equaler) bool {
+	// for the EnumType, we consider enclosed Values as an equality to
+	// allow changes to the template to be adding new values to the
+	// enum. If a default Equal() behaviour is needed, use DefaultEqual().
+	return t.EqualEnclosing(u)
```

Comment:
We should not say that two things are equal when they're not. Instead I recommend to add a special `EnumType` check here: 

https://github.com/fabric8-services/fabric8-wit/blob/8b6c84858e16aafc922c89577c47965c50b88328/spacetemplate/importer/repository.go#L160

The referenced section can be replaced with this, potentially outsourcing the compatibility check into its own `EnumType` member function:

```go
if equal := fd.Type.Equal(originalType); !equal {
	// Special treatment for EnumType
	origEnum, ok1 := originalType.(EnumType)
	newEnum, ok2 := fd.Type.(EnumType)
	if ok1 && ok2 {
		// check if only values are different
		// this could be outsourced into a member function of the EnumType
		tmpVals := origEnum.Values
		origValues.Values = newEnum.Values
		if origEnum.Equal(newEnum) {
			// values differ, so now check if the new ones include the old ones
			toBeFound := map[interface{}]struct{}{}
			for _, v := range tmpVals {
				toBeFound[v] = struct{}{}
			}
			for _, v := range newEnum.Values {
				delete(toBeFound, v)
			}
			equal = (len(toBeFound) == 0)
		}
	}
	if !equal {
		return errs.Errorf("type of the field %s changed from %s to %s", fieldName, originalType, fd.Type)
	}
}
```

Reply:
Fixed. 1b6cc50.